model/messages/synchronization.go

```diff
@@ -47,9 +48,44 @@ type BlockResponse struct {
 	Blocks []*flow.Block
 }
 
-// ClusterBlockResponse is the same thing as BlockResponse, but for cluster
-// consensus.
+var _ model.StructureValidator = (*BlockResponse)(nil)
+
+// StructureValid checks basic structural validity of the message and ensures no required fields are nil.
+func (m *BlockResponse) StructureValid() error {
+	for _, b := range m.Blocks {
```

Comment:
Should we allow an empty range? I think we don't want to process empty range because it doesn't make sense

Reply:
Can we somehow compose those checks so we use `b.StructureValid()` ?
I feel like we would benefit a lot from such recursive approach

Reply:
Good idea, added in https://github.com/onflow/flow-go/pull/3470/commits/8b47a64b24ac42d4f3df2b7c2a4ed1f6cacec715 and https://github.com/onflow/flow-go/pull/3470/commits/fc1605225f92c920ce630bcaa37eb4050d8fe957